### PR TITLE
Improve error message for sudo flatpak run

### PR DESCRIPTION
**Pull request by devurandom**
Closes: https://github.com/flatpak/flatpak/issues/3963

### CODE REVIEW

**Comment by mwleeds** [comment]
@devurandom you tried both `sudo -i flatpak run ...` and `su  -l flatpak run ...` to check that they work as intended?

**Comment by mwleeds** [comment]
For me the former works albeit with some error messages from glimpse and the latter does not work:
```
0 ✓ mwleeds@localhost ~ (1 key)
↳ sudo -i flatpak run org.glimpse_editor.Glimpse 
Gtk-Message: 14:27:49.724: Failed to load module "pk-gtk-module"
Error spawning command line “dbus-launch --autolaunch=b67a6021a5e14d43a25e3768ad03411c --binary-syntax --close-stderr”: Child process exited with code 1
Gtk-Message: 14:27:58.234: Failed to load module "pk-gtk-module"

(glimpse:2): GLib-GObject-CRITICAL **: 14:28:50.515: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
```
```
0 ✓ mwleeds@localhost ~ (1 key)
↳ su -l flatpak run org.glimpse_editor.Glimpse 
Password: 
su: Authentication failure
```

File: common/flatpak-run.c [hunk 1]
```diff
@@ -3514,7 +3514,7 @@ check_sudo (GError **error)
   /* SUDO_COMMAND could be a value like `/usr/bin/flatpak run foo` */
   split_command = g_strsplit (sudo_command_env, " ", 2);
   if (g_str_has_suffix (split_command[0], "flatpak"))
-    return flatpak_fail_error (error, FLATPAK_ERROR, _("\"flatpak run\" is not intended to be ran with sudo"));
+    return flatpak_fail_error (error, FLATPAK_ERROR, _("\"flatpak run\" is not intended to be ran with `sudo`, use `sudo -i` or `su -l` instead"));
```
Comment (mwleeds) [inline common/flatpak-run.c:3517]: Should be `run` not `ran` I think

Comment (devurandom) [reply common/flatpak-run.c:3517]:
```suggestion
    return flatpak_fail_error (error, FLATPAK_ERROR, _("\"flatpak run\" is not intended to be run with `sudo`, use `sudo -i` or `su -l` instead"));
```

**Comment by devurandom** [comment]
> @devurandom you tried both `sudo -i flatpak run ...` and `su -l flatpak run ...` to check that they work as intended?

No. I tried `sudo -i` and `su -l`. I am not sure about the exact effects of the former in this context. (Maybe the login shell when started without `-c` does something different than with `-c`. I did not check that or think much about it.) The latter is definitely wrong, as it tries to change to user `flatpak`, which probably does not exist or is not allowed to log in. Maybe I should clarify the exact way to execute the commands in the error message...

**Comment by mwleeds** [comment]
> > @devurandom you tried both `sudo -i flatpak run ...` and `su -l flatpak run ...` to check that they work as intended?
> 
> No. I tried `sudo -i` and `su -l`. I am not sure about the exact effects of the former in this context. (Maybe the login shell when started without `-c` does something different than with `-c`. I did not check that or think much about it.) The latter is definitely wrong, as it tries to change to user `flatpak`, which probably does not exist or is not allowed to log in. Maybe I should clarify the exact way to execute the commands in the error message...

Seems alright as is since you tested it

File: common/flatpak-run.c [hunk 2]
```diff
@@ -3514,7 +3514,7 @@ check_sudo (GError **error)
   /* SUDO_COMMAND could be a value like `/usr/bin/flatpak run foo` */
   split_command = g_strsplit (sudo_command_env, " ", 2);
   if (g_str_has_suffix (split_command[0], "flatpak"))
-    return flatpak_fail_error (error, FLATPAK_ERROR, _("\"flatpak run\" is not intended to be ran with sudo"));
+    return flatpak_fail_error (error, FLATPAK_ERROR, _("\"flatpak run\" is not intended to be run with `sudo`, use `sudo -i` or `su -l` instead"));
```
Comment (devurandom) [inline common/flatpak-run.c:3517]:
```suggestion
    return flatpak_fail_error (error, FLATPAK_ERROR, _("\"flatpak run\" is not intended to be run as `sudo flatpak run`, use `sudo -i` or `su -l` instead and invoke \"flatpak run\" from inside the new shell"));
```

Comment (smcv) [reply common/flatpak-run.c:3517]: I'm not @mwleeds, but I like that wording better than what's currently in this PR.

**Review by smcv — APPROVED** [review]

**Comment by smcv** [comment]
If @mwleeds doesn't object then let's merge this.

**Review by mwleeds — APPROVED** [review]

**Comment by sunshinewithmoonlight** [comment]
After I installed `io.github.Fndroid.clash_for_windows` with command `flatpak --system install io.github.Fndroid.clash_for_windows`, I get an error when I run `sudo -i flatpak  run io.github.Fndroid.clash_for_windows` :
```
[2 zypak-helper] Failed to connect to session bus: [org.freedesktop.DBus.Error.NotSupported] Using X11 for dbus-daemon autolaunch was disabled at compile time, set your DBUS_SESSION_BUS_ADDRESS instead
[2 zypak-helper] src/helper/main.cc:40(DetermineZygoteStrategy): Assertion failed: bus
```
Why???

**Comment by smcv** [comment]
This is a closed pull request from almost 2 years ago, not a technical support forum.

> `sudo -i flatpak  run io.github.Fndroid.clash_for_windows`

I don't know what you expect this command to do, but what it does is almost certainly not what you intend to be doing. Please run apps as an ordinary user (without `sudo`) instead.

**Comment by sunshinewithmoonlight** [comment]
Sorry for disturbing. However, I have to use root privileges when using this software because this software needs to install a system level service to work better.

**Pull request merged**